01_fib/davidbroughsmyth/main.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+// Function to calculate Fibonacci or Negafibonacci number
+// Range of n from -92 .. 0 .. 92
+func fib(n int) {
+
+	if n == 0 {
+		fmt.Fprintln(out, 0)
+		return
+	}
+
+	if n > 92 || n < -92 {
+		fmt.Fprintf(out, "Range for fib(n) should be between -92 to 92\n")
```

Comment:
As a mathematician I would point out that technically `-92 to 92` is the domain, or the valid range of `n`, not `fib(n)` 😉

Reply:
Good point, I was thinking the range of n but it does read for function fib(n).